browser/brave_content_browser_client.cc

```diff
@@ -236,6 +241,26 @@ void BindCosmeticFiltersResources(
       std::move(receiver));
 }
 
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+void BindBraveWalletProvider(
+    content::RenderFrameHost* const frame_host,
+    mojo::PendingReceiver<brave_wallet::mojom::BraveWalletProvider> receiver) {
+  auto* web_contents = content::WebContents::FromRenderFrameHost(frame_host);
+  if (!web_contents && web_contents->GetBrowserContext()->IsTor())
+    return;
+
+  auto* profile =
+      Profile::FromBrowserContext(web_contents->GetBrowserContext());
+  BraveWalletService* service =
+      BraveWalletServiceFactory::GetInstance()->GetForProfile(
+          Profile::FromBrowserContext(profile));
+
+  mojo::MakeSelfOwnedReceiver(
+      std::make_unique<brave_wallet::BraveWalletProvider>(service),
```

Comment:
I don't think it's safe to pass the service here like this because I think it's possible that an ipc message could come in after the service has been destroyed. I think you need to pass in a WeakPtr ref here

Reply:
wdyt @iefremov ? The renderer processes should all shut down before the profile (and service) is destroyed, but could there still be an unprocessed message in the queue?

Reply:
Made is as a WeakPtr to be safe.